moveit_kinematics/kdl_kinematics_plugin/src/kdl_kinematics_plugin.cpp

```diff
@@ -478,12 +478,14 @@ int KDLKinematicsPlugin::CartToJnt(KDL::ChainIkSolverVelMimicSVD& ik_solver, con
                     delta_q_norm);
     if (delta_q_norm < epsilon_)  // stuck in singularity
     {
-      if (step_size < 0.005)  // cannot reach target
+      if (step_size < epsilon_)  // cannot reach target
         break;
       // wiggle joints
       last_delta_twist_norm = DBL_MAX;
       delta_q.data.setRandom();
       delta_q.data *= std::min(0.1, delta_twist_norm);
+      clipToJointLimits(q_out, delta_q, extra_joint_weights);
+      extra_joint_weights.setOnes();
```

Comment:
Is it justified to throw away the values of `extra_joint_weights`, computed in the call to `clipToJointLimits` above?
Intuitively, I would have expected a separate vector to pass here.

Reply:
When we wiggle the joints, we essentially start over:
- `last_delta_twist_norm` is reset
- as well as `extra_joint_weights`

Reply:
makes sense, I did not notice the norm reset.